pkg/apis/devops/v1alpha3/devopsproject_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The KubeSphere authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const DevOpsProjectFinalizerName = "devopsproject.finalizers.kubesphere.io"
+
+const (
+	ResourceKindDevOpsProject     = "DevOpsProject"
+	ResourceSingularDevOpsProject = "devopsproject"
+	ResourcePluralDevOpsProject   = "devopsprojects"
+)
+
+// DevOpsProjectSpec defines the desired state of DevOpsProject
+type DevOpsProjectSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+}
+
+// DevOpsProjectStatus defines the observed state of DevOpsProject
+type DevOpsProjectStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DevOpsProject is the Schema for the devopsprojects API
+// +kubebuilder:resource:categories="devops",scope="Cluster"
+// +k8s:openapi-gen=true
+type DevOpsProject struct {
```

Comment:
Since DevOpsProject is in group devops.kubesphere.io, it seems a little bit unnecessary for extra devops in the name.  Or you are trying to make it more specific to distinguish  with other projects?

Reply:
I think that project is too generic. When using kubectl, it may have the same name as other CRDs (although they are in different apigroups).